src/Containers/Timeline.tsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import styled from 'styled-components';
+import { flex } from '../Utils/Mixins';
+import {
+    Responsive,
+    MainInterface,
+    ResponsiveInterface,
+} from '../Utils/BaseStyles';
+
+interface DataItem {
+    color: string;
+    Icon: React.FC<React.HTMLAttributes<HTMLDivElement>>;
+    text: string;
+}
+
+export interface TimelineProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    title: string;
+    titleColor: string;
+    subContentLeftTitle: string;
+    subContentLeftAmount: string | number;
+    subContentRightTitle: string;
+    subContentRightAmount: string | number;
+    figuresColor: string;
+    textColor: string;
+    separatorColor: string;
+    separatorLength: string | number;
+    data: Array<DataItem>;
+}
+
+export const Timeline: React.FC<TimelineProps> = ({
+    title,
+    titleColor,
+    subContentLeftTitle,
+    subContentLeftAmount,
+    subContentRightTitle,
+    subContentRightAmount,
+    figuresColor,
+    textColor,
+    separatorColor,
+    separatorLength,
+    data,
+    ...props
+}): React.ReactElement => {
+    const subContentLeftAmountLocale = `$${subContentLeftAmount.toLocaleString()}`;
+    const subContentRightAmountLocale = `$${subContentRightAmount.toLocaleString()}`;
+
+    const getTimelineItem = (Item: DataItem, index: number) => (
+        <TimelineItem key={`${index}_timeline`}>
+            <TimelineItemLeft color={separatorColor} isFirst={index === 0}>
+                <Item.Icon style={{ width: '15px', color: Item.color }} />
```

Comment:
This specifically with the prop of Icon: React.FC<React.HTMLAttributes<HTMLDivElement>> is not correct, plus inline styling. You are passing in a StyledIcon in storybook which is React.ForwardRefExoticComponent<React.RefAttributes<SVGSVGElement>>, or more specifically import { StyledIcon } from 'styled-icons/types'.  This should be a styled SVG that accepts props of color, and use the format as={Item.Icon}